src/ws/test-channelresponse.c

```diff
@@ -541,6 +541,15 @@ test_resource_failure (TestResourceCase *tc,
   g_assert (cockpit_pipe_get_pid (tc->pipe, &pid));
   g_assert_cmpint (pid, >, 0);
   int pid_fd = syscall(SYS_pidfd_open, pid, 0);
+  if (pid_fd < 0)
+    {
+      if (errno == ENOSYS)
+	g_test_skip ("no pidfd_open support, skipping");
+      else
+        g_test_skip (strcat("pidfd_open call failed: ", g_strerror (errno)));
```

Comment:
We shouldn't silently skip the test if that fails for any other reason, as then we have screwed up the code. Also, the strcat() here is both invalid (you can't append to a const string), and if it *would* copy, it'd be a memory leak. To fix all of these, use this:

    g_error ("pidfd_open call failed: %m");

Reply:
Ah yes, I was not 100% certain how we would like to handle it. `g_error` makes sense.